core/src/appleMain/kotlin/CentralManager.kt

```diff
@@ -6,25 +6,42 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.withContext
 import platform.CoreBluetooth.CBCentralManager
+import platform.CoreBluetooth.CBCentralManagerOptionRestoreIdentifierKey
 import platform.CoreBluetooth.CBCharacteristic
 import platform.CoreBluetooth.CBCharacteristicWriteType
 import platform.CoreBluetooth.CBDescriptor
 import platform.CoreBluetooth.CBPeripheral
 import platform.CoreBluetooth.CBService
 import platform.CoreBluetooth.CBUUID
 import platform.Foundation.NSData
+import platform.Foundation.NSUUID
+import platform.Foundation.NSUserDefaults
 
 private const val DISPATCH_QUEUE_LABEL = "central"
+private const val CBCENTRALMANAGER_RESTORATION_ID = "kable-central-manager"
+private const val CBCENTRALMANAGER_CONSUMER_ID_KEY = "kable-central-manager-consumer-id"
 
 public class CentralManager internal constructor() {
 
     internal companion object {
         val Default: CentralManager by lazy { CentralManager() }
     }
 
+    private val userDefaults = NSUserDefaults.standardUserDefaults
+
+    // This value is needed to ensure multiple instances of Kable running on the same iOS device
+    // do not cross pollinate restored instances of CBCentralManager. The value will live for the
+    // lifetime of the consuming app.
+    private val consumerId: String
+        get() = userDefaults.stringForKey(CBCENTRALMANAGER_CONSUMER_ID_KEY)
+            ?: NSUUID().UUIDString().also {
+                userDefaults.setObject(it, CBCENTRALMANAGER_CONSUMER_ID_KEY)
+            }
+
     private val dispatcher = QueueDispatcher(DISPATCH_QUEUE_LABEL)
     internal val delegate = CentralManagerDelegate()
-    private val cbCentralManager = CBCentralManager(delegate, dispatcher.dispatchQueue)
+    private val cbOptions = mutableMapOf<Any?, Any>(CBCentralManagerOptionRestoreIdentifierKey to CBCENTRALMANAGER_RESTORATION_ID + consumerId).toMap()
+    private val cbCentralManager = CBCentralManager(delegate, dispatcher.dispatchQueue, cbOptions)
```

Comment:
It doesn't look necessary to have the mutable map, nor the typing defined? This could be simplified as:

```kotlin
    private val cbCentralManager = CBCentralManager(
        delegate,
        dispatcher.dispatchQueue,
        options = mapOf(CBCentralManagerOptionRestoreIdentifierKey to "$CBCENTRALMANAGER_RESTORATION_ID-$consumerId"),
    )
```